R_scripts/sapflow.Rmd

```diff
@@ -0,0 +1,471 @@
+  summarise(failures = sum(!good_cycle, na.rm = TRUE), .groups = "drop") ->
+  DAILY_TESTS[["WARN.daily_cycle"]]
+
+sapflow %>% 
+  left_join(daily_amplitude_data, by = c("Date", "Tree_Code", "Port")) ->
+  sapflow
+
+p_dc <- ggplot(sapflow, aes(Timestamp, Value, color = good_cycle)) + 
+  geom_point(size = 0.5) + 
+  facet_wrap(~Tree_Code) + 
+  coord_cartesian(ylim = c(0, 1)) +
+  scale_color_manual(values = TEST_COLOR_SCALE) +
+  theme(axis.text.x = element_text(angle = 90))
+```
+
+### Control
+
+```{r}
+p_dc %+% filter(sapflow, Plot == "C")
+```
+
+### Fresh
+
+```{r}
+p_dc %+% filter(sapflow, Plot == "F")
+```
+
+### Salt
+
+```{r}
+p_dc %+% filter(sapflow, Plot == "S")
+```
+
+### Shore
+
+```{r}
+p_dc %+% filter(sapflow, Plot == "L")
+```
+
+## Trend test {.tabset}
+
+```{r}
+TREND_DAYS <- 7
+```
+
+This test looks at every `r TREND_DAYS` days of data and tests whether there's a significant 
+trend (using a Mann-Kendall test) in the daily mean.
+
+```{r trend-test}
+sapflow %>% 
+  mutate(group = yday(Timestamp) %/% TREND_DAYS) ->
+  sapflow
+
+library(Kendall)
+
+mk_no_trend <- function(x) {
+  if(length(x) < 3) NA else {
+    capture.output(mk <- MannKendall(x))
+    mk$sl > 0.05
+  }
+}
+
+sapflow %>% 
+  arrange(Date) %>% 
+  group_by(Date, group, Tree_Code, Port) %>% 
+  summarise(Value = mean(Value), .groups = "drop") %>% 
+  group_by(group, Tree_Code, Port) %>% 
+  summarise(no_trend = mk_no_trend(Value),
+            .groups = "drop") ->
+  weekly_trends
+
+# Store results
+weekly_trends %>% 
+  group_by(Tree_Code, Port) %>%
+  summarise(failures = sum(!no_trend, na.rm = TRUE), .groups = "drop") ->
+  DAILY_TESTS[["WARN.trend"]]
+
+sapflow %>% 
+  left_join(weekly_trends, by = c("group", "Tree_Code", "Port")) ->
+  sapflow
+
+p_tt <- ggplot(sapflow, aes(Timestamp, Value, color = no_trend)) + 
+  geom_point(size = 0.5) + 
+  facet_wrap(~Tree_Code) + 
+  coord_cartesian(ylim = c(0, 1)) +
+  scale_color_manual(values = TEST_COLOR_SCALE) +
+  theme(axis.text.x = element_text(angle = 90))
+```
+
+### Control
+
+```{r}
+p_tt %+% filter(sapflow, Plot == "C")
+```
+
+### Fresh
+
+```{r}
+p_tt %+% filter(sapflow, Plot == "F")
+```
+
+### Salt
+
+```{r}
+p_tt %+% filter(sapflow, Plot == "S")
+```
+
+### Shore
+
+```{r}
+p_tt %+% filter(sapflow, Plot == "L")
+```
+
+# Summary
+
+## Daily test failure count:
+
+```{r summary, warning=FALSE, fig.width = 7, fig.height = 8}
+DAILY_TESTS %>% 
+  bind_rows(.id = "test") %>% 
+  left_join(sf_inventory, by = c("Tree_Code", "Port")) %>% 
+  select(test, Plot, Tree_Code, Species, Port, 
+         Grid_Square, Logger, Installation_Date, failures) %>% 
+  mutate(Grid_Letter = substring(Grid_Square, 1, 1), 
+         Grid_Number = substring(Grid_Square, 2, 2)) -> daily_test
+
+# Spatial map of errors
+ggplot(filter(daily_test, Plot %in% c("C", "F", "S")), aes(x = Grid_Letter, y = Grid_Number)) + 
+  geom_jitter(aes(color = test)) + facet_wrap(~Plot, ncol = 1) + theme_minimal() -> p
```

Comment:
If using `geom_jitter()` I would do `set.seed(...)` at the beginning of the script. That way the points won't randomly jump around when you re-run script with exact same data.